### PR TITLE
61 blog details page 11.04.2023

### LINKED ISSUE #61: Blog page item
**As a user,**
I want to see a Blog item page, a separate page for case study content. 

### Links to design:
- [Figma web](https://www.figma.com/file/Pp48CPTdPVHY3Fkrxqn45b/Varbin-Tech_?node-id=536-886&t=1fYz1InlZJu6sXQA-4)
- [Figma mob](https://www.figma.com/file/Pp48CPTdPVHY3Fkrxqn45b/Varbin-Tech_?node-id=613-22342&t=1fYz1InlZJu6sXQA-4)

### Requirements:
- The page should use the same layout as in the case of the Index page
- We should use [Dynamic Routes](https://nextjs.org/docs/routing/dynamic-routes)
- Data structure is up to you
- Background image for a Hero section should be provided via a special prop
- Share buttons shouldn't use any JavaScript. All of them should be plain HTML + CSS
- Instead of a GitHub button, add a Facebook button
- "Book a call" button is a link button which contains a link to Calendly (An example can be found on the Index page)
- "View more" button is a standard MUI button which loads three more posts after each click
- Rename "Related blogs" to "Related posts". This section should be a separate component
- "Table of Contents" section should be scrolling while a user is scrolling a post
- "Table of Contents" should contain all titles from the post
- When a user clicks on any title in the "Table of Contents", we should scroll the content to this title
- Tags line in the Hero section should behave as links

### PR DESCRIPTION
**Pull request by InnaLazarenko**
### List of tasks:
- resolved #61

### CODE REVIEW

File: components/common/table-of-content/TableOfContent.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      };
+    }
+
+    return;
+  });
+
+  return (
+    <TableOfContentContainer>
+      <ImageContainer>
+        <span className="inner-wrapper">
```
Comment (var-bin) [inline components/common/table-of-content/TableOfContent.tsx:36]: Please, use `ImageWrapper`

File: components/common/table-of-content/TableOfContent.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            width={image.width}
+            height={image.height}
+          />
+        </span>
+      </ImageContainer>
+      <Typography variant="h5" marginBottom={1.75}>
+        Table of Contents
+      </Typography>
+      <Stack spacing={2} marginBottom="52px">
+        {linkList &&
```
Comment (var-bin) [inline components/common/table-of-content/TableOfContent.tsx:51]: If replace `linkList &&` with `linkList.length &&`, the additional check `if (item && item.title) {` could be omitted.

File: components/common/table-of-content/TableOfContent.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+          linkList.map((item, index) => {
+            if (item && item.title) {
+              return (
+                <Link key={index} href={`#${item.href}`} underline="hover">
+                  <>{item.title}</>
```
Comment (var-bin) [inline components/common/table-of-content/TableOfContent.tsx:56]: Do we actually need to use Fragment (i.e. `<></>`) here?

File: pages/blog/[blogId].tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+          projectTags={data.blogTags}
+          projectSocialIcons={socialShareButtons(pageLink)}
+          postAuthor={data.blogAuthor}
+          postDate={data.blogDate}
+          postAuthorPhoto={data.blogAuthorPhoto}
+          postBgImage={data.blogImage.src}
+        />
+      ) : null}
+
+      {data ? (
```
Comment (var-bin) [inline pages/blog/[blogId].tsx:38]: It seems we can omit the second check and put the markup on the first one.

**Review by var-bin — CHANGES_REQUESTED** [review]

File: pages/blog/[blogId].tsx (hunk 4 above)
Comment (InnaLazarenko) [reply pages/blog/[blogId].tsx:38]: '+

File: components/common/table-of-content/TableOfContent.tsx (hunk 3 above)
Comment (InnaLazarenko) [reply components/common/table-of-content/TableOfContent.tsx:56]: '+

File: components/common/table-of-content/TableOfContent.tsx (hunk 2 above)
Comment (InnaLazarenko) [reply components/common/table-of-content/TableOfContent.tsx:51]: '+

**Review by var-bin — APPROVED** [review]

**Pull request merged**